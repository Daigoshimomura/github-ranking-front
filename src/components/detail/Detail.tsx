import React from "react"

type DetailProps = {
    foo: string;
  }
  export class Detail extends React.Component<DetailProps, {}> {
      render() {
          return <span>{this.props.foo}</span>
      }
  }
  