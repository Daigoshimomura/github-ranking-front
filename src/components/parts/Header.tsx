import React from "react";
import MenuIcon from "@material-ui/icons/Menu";
import { Link } from "react-router-dom";

import {
  AppBar,
  Toolbar,
  IconButton,
  Typography,
  ListItem,
  ListItemText,
  List,
  Drawer
} from "@material-ui/core";

export interface HeaderState {
  left: boolean;
}
interface HeaderProps {}

export class Header extends React.Component<HeaderProps, HeaderState> {
  constructor(props: HeaderProps) {
    super(props);
    this.state = {
      left: false
    };
  }

  toggleDrawer = (open: boolean) => () => {
    this.setState({
      left: open
    });
  };

  render() {
    const sideList = (
      <div>
        <List>
          <Link to="/">
            <ListItem button>
              <ListItemText primary="Back to Top" />
            </ListItem>
          </Link>
        </List>
      </div>
    );

    return (
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            color="inherit"
            aria-label="menu"
            onClick={this.toggleDrawer(true)}
          >
            <MenuIcon />
          </IconButton>
          <Drawer open={this.state.left} onClose={this.toggleDrawer(false)}>
            <div
              style={{ width: 145 }}
              tabIndex={0}
              role="button"
              onClick={this.toggleDrawer(false)}
              onKeyDown={this.toggleDrawer(false)}
            >
              {sideList}
            </div>
          </Drawer>
          <Typography variant="h6">GitHub Ranking</Typography>
        </Toolbar>
      </AppBar>
    );
  }
}
