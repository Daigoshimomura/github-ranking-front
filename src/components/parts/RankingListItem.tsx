import React from "react";
import { UserMinimumInformation } from "../../external/data/UserMinimumInformation";
import {
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText
} from "@material-ui/core";

interface RankingListItemProps {
  rank: Number;
  user: UserMinimumInformation;
  itemClickCallback: Function;
}

export class RankingListItem extends React.Component<RankingListItemProps, {}> {
  constructor(props: RankingListItemProps) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.itemClickCallback(this.props.user.userId);
  }

  render() {
    return (
      <ListItem onClick={this.handleClick} style={{ cursor: "pointer" }}>
        <ListItemAvatar>
          <Avatar
            alt={this.props.user.userId}
            src={this.props.user.avatarUrl}
          />
        </ListItemAvatar>
        <ListItemText primary={this.props.rank} />
        <ListItemText primary={this.props.user.userId} />
      </ListItem>
    );
  }
}
