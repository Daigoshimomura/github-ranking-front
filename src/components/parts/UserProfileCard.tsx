import { UserDetailRes } from "../../external/data/UserDetailRes";
import React from "react";
import { Card, CardMedia, CardContent, Typography } from "@material-ui/core";

interface UserProfileCardProps {
  userDetail: UserDetailRes;
}

export class UserProfileCard extends React.Component<UserProfileCardProps, {}> {
  render() {
    return (
      <Card style={{ maxWidth: 600, display: "flex" }}>
        <CardMedia
          src={this.props.userDetail.avatarUrl}
          component="img"
          style={{ width: 100, margin: 15 }}
        ></CardMedia>
        <CardContent>
          <Typography variant="h6" component="h6">
            ID : {this.props.userDetail.userId}
          </Typography>
          <Typography color="textSecondary">
            Rank : {this.props.userDetail.rank}/
            {this.props.userDetail.currentNumber}
          </Typography>
          <Typography color="textSecondary">
            Score : {this.props.userDetail.score}
          </Typography>
          <Typography color="textSecondary">
            MainLanguage : {this.props.userDetail.mainLanguage}
          </Typography>
        </CardContent>
      </Card>
    );
  }
}
