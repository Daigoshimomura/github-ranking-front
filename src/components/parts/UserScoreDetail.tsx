import React from "react";
import { UserDetailRes } from "../../external/data/UserDetailRes";
import {
  TableContainer,
  Table,
  TableHead,
  TableRow,
  TableCell,
  Paper,
  TableBody
} from "@material-ui/core";

interface UserScoreDetailPorps {
  userDetail: UserDetailRes;
}

export class UserScoreDetail extends React.Component<UserScoreDetailPorps, {}> {
  render() {
    return (
      <TableContainer component={Paper}>
        <Table aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>Detail Score</TableCell>
              <TableCell align="right">followers</TableCell>
              <TableCell align="right">issues</TableCell>
              <TableCell align="right">pullRequests</TableCell>
              <TableCell align="right">repositories</TableCell>
              <TableCell align="right">forks</TableCell>
              <TableCell align="right">stargazer</TableCell>
              <TableCell align="right">watchers</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell component="th" scope="row"></TableCell>
              <TableCell align="right">
                {this.props.userDetail.followersCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.issuesCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.pullRequestCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.repositoriesCount}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.forksCountTotal}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.stargazerCountTotal}
              </TableCell>
              <TableCell align="right">
                {this.props.userDetail.watchersCountTotal}
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </TableContainer>
    );
  }
}
