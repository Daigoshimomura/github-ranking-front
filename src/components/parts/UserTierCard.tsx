import React from "react";
import { Card, CardMedia, CardContent, Typography } from "@material-ui/core";

interface UserTierCardProps {
  tierName: string;
  tierAbout: string;
  imageSrc: string;
  score: Number;
}

export class UserTierCard extends React.Component<UserTierCardProps, {}> {
  render() {
    return (
      <Card style={{ maxWidth: 270, display: "flex" }}>
        <CardMedia
          image={this.props.imageSrc}
          component="img"
          style={{ width: 100, margin: 15 }}
        ></CardMedia>
        <CardContent>
          <Typography color="primary" variant="h6" component="h6">
            {this.props.tierName}
          </Typography>
          <Typography color="textSecondary">
            Score : {this.props.score}
          </Typography>
          <Typography color="textSecondary">{this.props.tierAbout}</Typography>
        </CardContent>
      </Card>
    );
  }
}
