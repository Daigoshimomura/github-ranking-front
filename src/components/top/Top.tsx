import React from "react"

type TopProps = {
    foo: string;
  }
 export class Top extends React.Component<TopProps, {}> {
      render() {
          return <span>{this.props.foo}</span>
      }
  }
  