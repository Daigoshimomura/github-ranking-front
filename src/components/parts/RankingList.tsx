import React from "react";
import { RanksByLanguage } from "../../external/data/RanksByLanguage";
import { Typography, Grid, List } from "@material-ui/core";
import { RankingListItem } from "./RankingListItem";

interface RankingListProps {
  ranksByLanguage: RanksByLanguage;
  itemClickCallback: Function;
}

export class RankingList extends React.Component<RankingListProps, {}> {
  render() {
    return (
      <Grid item xs={12} md={4}>
        <Typography variant="h6">
          {this.props.ranksByLanguage.language}
        </Typography>
        <div>
          <List>
            {this.props.ranksByLanguage.userInfomations.map((t, index) => (
              <RankingListItem
                key={t.userId}
                user={t}
                rank={index + 1}
                itemClickCallback={this.props.itemClickCallback}
              />
            ))}
          </List>
        </div>
      </Grid>
    );
  }
}
