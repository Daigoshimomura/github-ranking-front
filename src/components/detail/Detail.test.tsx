import { UNSEARCHED } from "../../const/UtilCont";
import ControllUser from "../../external/ControllUser";
import { Detail } from "./Detail";
import { mount } from "enzyme";
import React from "react";
import { routerTestProps } from "../../util/test/RoutComponentMock";
import { createLocation, createMemoryHistory } from "history";
import { TopState } from "../top/Top";
import { CircularProgress } from "@material-ui/core";
import { NOTFOUND } from "dns";
import { UserProfileCard } from "../parts/UserProfileCard";
import renderer from "react-test-renderer";

function setUp(state: TopState) {
  const { match } = routerTestProps("/route/:userId", {
    userId: "testUserId"
  });
  const location = createLocation<TopState>(match.url);
  location.state = state;
  const history = createMemoryHistory<TopState>();

  const wrapper = mount<Detail>(
    <Detail
      userId={"userId"}
      match={match}
      location={location}
      history={history}
    />
  );

  return wrapper;
}

describe("Detail", () => {
  it("snapshot", () => {
    const { match } = routerTestProps("/route/:userId", {
      userId: "testUserId"
    });
    const location = createLocation<TopState>(match.url);
    location.state = {
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    };
    const history = createMemoryHistory<TopState>();
    const tree = renderer
      .create(
        <Detail
          userId={"userId"}
          match={match}
          location={location}
          history={history}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("レンダリングの確認", () => {
    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });

    //ロード中
    expect(wrapper.find(CircularProgress).length).toBe(1);

    wrapper.setState({
      userDetail: {
        userName: "",
        userId: NOTFOUND,
        avatarUrl: "",
        githubUrl: "",
        tier: "",
        rank: 0,
        score: 0,
        currentNumber: 0,
        followersCount: 0,
        issuesCount: 0,
        pullRequestCount: 0,
        repositoriesCount: 0,
        forksCountTotal: 0,
        stargazerCountTotal: 0,
        watchersCountTotal: 0,
        mainLanguage: "",
        lastupdateDate: new Date("2019/01/01")
      }
    });

    // Not Found
    expect(wrapper.find(".backToTopButton").length).toBeGreaterThan(1);

    wrapper.setState({
      userDetail: {
        userName: "",
        userId: "userId",
        avatarUrl: "",
        githubUrl: "",
        tier: "",
        rank: 0,
        score: 0,
        currentNumber: 0,
        followersCount: 0,
        issuesCount: 0,
        pullRequestCount: 0,
        repositoriesCount: 0,
        forksCountTotal: 0,
        stargazerCountTotal: 0,
        watchersCountTotal: 0,
        mainLanguage: "",
        lastupdateDate: new Date("2019/01/01")
      }
    });

    // 見つかった場合
    expect(wrapper.find(UserProfileCard).length).toBe(1);
  });
  it("propsをちゃんと渡せてることの確認、componentDidMountの確認", () => {
    const getUserDetailspy = jest.spyOn(Detail.prototype, "getUserDetail");
    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    // propsの確認
    expect(wrapper.prop("userId")).toBe("userId");
    // componentDidMount の確認
    expect(getUserDetailspy).toHaveBeenCalled();
  });

  it("getUserDetailの確認", () => {
    const getUser = jest.fn().mockImplementation(() => Promise.resolve({}));
    ControllUser.getUserData = getUser;

    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    expect(wrapper.prop("userId")).toBe("userId");
    expect(getUser).toHaveBeenCalled();
  });

  it("onClickUpdateButtonの確認", () => {
    // メソッド呼び出しのモック化
    const updateUserDetail = jest.spyOn(Detail.prototype, "updateUserDetail");

    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    const comp: Detail = wrapper.instance();
    comp.setState({
      userDetail: {
        userName: "",
        userId: UNSEARCHED,
        avatarUrl: "",
        githubUrl: "",
        tier: "",
        rank: 0,
        score: 0,
        currentNumber: 0,
        followersCount: 0,
        issuesCount: 0,
        pullRequestCount: 0,
        repositoriesCount: 0,
        forksCountTotal: 0,
        stargazerCountTotal: 0,
        watchersCountTotal: 0,
        mainLanguage: "",
        lastupdateDate: new Date("2019/01/01")
      }
    });
    // 異なる日ならコールされる
    comp.onClickUpdateButton();
    expect(updateUserDetail).toHaveBeenCalled();

    comp.setState({
      userDetail: {
        userName: "",
        userId: UNSEARCHED,
        avatarUrl: "",
        githubUrl: "",
        tier: "",
        rank: 0,
        score: 0,
        currentNumber: 0,
        followersCount: 0,
        issuesCount: 0,
        pullRequestCount: 0,
        repositoriesCount: 0,
        forksCountTotal: 0,
        stargazerCountTotal: 0,
        watchersCountTotal: 0,
        mainLanguage: "",
        lastupdateDate: new Date()
      }
    });
    // 同日ならopenする
    comp.onClickUpdateButton();
    expect(comp.state.openDialog).toBe(true);
  });

  it("updateUserDetailの確認", () => {
    const updateUser = jest.fn().mockImplementation(() => Promise.resolve({}));
    ControllUser.updateUserData = updateUser;

    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    const comp: Detail = wrapper.instance();
    comp.updateUserDetail("userId");
    expect(updateUser).toHaveBeenCalled();
  });

  it("backTopの確認", () => {
    const mock = jest.fn();

    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    const comp: Detail = wrapper.instance();
    comp.props.history.push = mock;
    comp.backTop();
    expect(mock).toHaveBeenCalled();
  });

  it("backTopの確認", () => {
    const wrapper = setUp({
      userId: "userId",
      dialogOpen: false,
      ranking: {
        rankByLanguages: []
      }
    });
    const comp: Detail = wrapper.instance();

    comp.setState({
      openDialog: true
    });

    comp.dialogClose();
    expect(comp.state.openDialog).toBe(false);
  });
});
