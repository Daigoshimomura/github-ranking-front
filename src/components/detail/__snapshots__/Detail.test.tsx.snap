// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root MuiTypography-h6"
        >
          GitHub Ranking
        </h6>
      </div>
    </header>
    <div>
      <div
        className="MuiBox-root MuiBox-root-117"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
        >
          Now Loading...
        </h6>
      </div>
      <div
        className="MuiBox-root MuiBox-root-118"
      >
        <div
          className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
